doc/srv6/srv6-hld-v19.md

```diff
@@ -0,0 +1,552 @@
+segment = SRv6_SID_LIST.key,       ; List of segment names for END.B6.ENCAP, can be empty
+source  = address,                 ; List of src addrs for encap for END.B6.ENCAP
+weight  = weight.value,            ; List of assigned weight, can be empty
+```
+
+**Modify ROUTE_TABLE**
+
+Description: Existing Route Table is extended to add SID list. 
+
+Schema:
+
+```
+;Stores a list of routes
+;Status: Mandatory
+
+key = ROUTE_TABLE:VRF_NAME:prefix ;
+nexthop = prefix,              ; IP addresses separated ',' (empty indicates no gateway). May indicate VXLAN endpoint if vni_label is non zero.
+intf = ifindex? PORT_TABLE.key ; zero or more separated by ',' (zero indicates no interface)
+vni_label = VRF.vni            ; zero or more separated by ',' (empty value for non-vxlan next-hops). May carry MPLS label in future.
+router_mac = mac_address       ; zero or more remote router MAC address separated by ',' (empty value for non-vxlan next-hops)
+blackhole = BIT                ; Set to 1 if this route is a blackhole (or null0)
+weight = weight_list           ; List of weights
+sid = SRV6_STEER_TABLE.key,    ; New optional field. List of SID, separated by ',' 
+seg_src = address              ; New optional field. Source addrs for sid encap
+```
+
+## 2.3 Orchestration Agent Changes
+
+New Orchagent(SRV6Orch) is created to manage all SRV6 related objects. SRV6Orchagent listens to APP_DB for regular updates and create/update SAI objects in ASIC_DB.  
+
+![draw-configdb](images/srv6db.png)
+
+![draw-configdb](images/srv6orch.png)
+
+**SRV6Orchagent**
+
+This orchagent is responsible to create SRV6 related objects in ASIC_DB with the information from APP_DB.
+
+
+
+SRV6Orchagent listens to all updates of SRV6_SID_LIST_TABLE to create SAI_SEGMENTROUTE_SIDLIST object with list of V6 prefixes. It also creates a SRV6 Nexthop with the existing SIDLIST object handle. Any update to V6 prefixes to the segment will be pushed to ASIC_DB.
+
+
+
+When a route entry is added to ROUTE_TABLE, routeOrchagent calls srv6Orchagent to get the SRV6 nexthop with all associated segment prefixes. If a route entry is referenced by list of ECMP segments, the orchagent creates a ECMP group with already created Nexthop members and pass the ECMP object handle to routeOrchagent. When all route entries referenced by the ECMP groups are deleted, ECMP group object is deleted. 
+
+
+
+Orchagent listens to LOCAL_SID_TABLE in APP_DB to create SAI objects in ASIC_DB. For LOCAL_SID's END.X action, this orchagent queries the existing IP NextHop and NextHopGroup database and use the existing object handle and update ASIC_DB. This orchagent creates a new ECMP group when Nexthop exists for all the Nexthop addresses in END.X action and no matching group exists in the DB. For LOCAL_SID's END.DT46 action, orchagent passes the VRF handle associated with VRF name to ASIC_DB. For LOCAL_SID's END.B6 Encaps, orchagent use existing Nexthop/NexthopGroup for the list of segments or create a new NexthopGroup.
```

Comment:
If the IP next-hop is not resolved yet, we should follow the proactive resolution for NH we use today?

Reply:
Done. Updated section with NH resolution.